server/database/postgres.go

```diff
@@ -92,11 +115,26 @@ func CheckAgent(idUser string, idAgent string) (bool, error) {
 	return agent.ID != "", nil
 }
 
+// GetAgentOwner returns user ID registered for particular agent
+func GetAgentOwner(agentID string) (string, error) {
+	agent := Agent{}
+	err := db.Where("agents.id = ?", agentID).Find(&agent).Error
+	// If agent has no user id, thats ok, user will sign up soon
+	if err == gorm.ErrRecordNotFound {
+		return "", nil
+	}
+
+	if err != nil {
+		return "", err
+	}
+	return agent.UserID, nil
+}
+
 //RegisterNewAgent adds a new agent to user, returns nil if adding was successful
-func RegisterNewAgent(idUser string, idAgent string) error {
-	agent := Agent{ID: idAgent, UserID: idUser}
+func RegisterNewAgent(idAgent string) error {
+	agent := Agent{ID: idAgent}
 	rows := db.Create(&agent).RowsAffected
```

Comment:
You save agent without user.

Reply:
yes, when agent connects, there is no entry for user in database. We should discuss it tomorrow